tests/daikon-tests/BinaryTree/BinaryTree.txt-merge-esc.goal

```diff
@@ -30,10 +30,10 @@ public class BinarySearchTree<E extends Comparable>
     root = null;
   }
 
-  /*@ requires x != null; */
   /*@ requires \typeof(x) == \type(DataStructures.MyInteger); */
   /*@ modifies this.root; */
   /*@ ensures this.root != null; */
+  /*@ ensures \old(x) != null; */
```

Comment:
This is an odd change.  I find the original line clearer and more intuitive than the new one.
Do you understand the cause of this change in output?  What would it take to restore the behavior?

Reply:
Sorry, I do not understand the cause of this  change.

Reply:
OK, we need to understand this before we merge the change.